src/ukify/ukify.py

```diff
@@ -1306,16 +1317,15 @@ def make_uki(opts: UkifyConfig) -> None:
         cmd = [keyutil_tool, 'public']
 
         if opts.pcr_public_keys and len(opts.pcr_public_keys) == 1:
+            pcrpkey = Path(opts.pcr_public_keys[0])
+        elif opts.pcr_certificates and len(opts.pcr_certificates) == 1:
             # If we're using an engine or provider, the public key will be an X.509 certificate.
-            if opts.signing_engine or opts.signing_provider:
-                cmd += ['--certificate', opts.pcr_public_keys[0]]
-                if opts.certificate_provider:
-                    cmd += ['--certificate-source', f'provider:{opts.certificate_provider}']
+            cmd += ['--certificate', opts.pcr_certificates[0]]
+            if (opts.signing_engine or opts.signing_provider) and opts.certificate_provider:
+                cmd += ['--certificate-source', f'provider:{opts.certificate_provider}']
 
-                print('+', shell_join(cmd), file=sys.stderr)
-                pcrpkey = subprocess.check_output(cmd)
-            else:
-                pcrpkey = Path(opts.pcr_public_keys[0])
+            print('+', shell_join(cmd), file=sys.stderr)
+            pcrpkey = subprocess.check_output(cmd)
```

Comment:
AFAIK this is a compat break. While I think the current logic of interpreting the public key as a certificate if an engine or provider is passed is rather horrifying, we have to keep it intact for compat.

So instead I'd just keep this as is and then add another `elif` at the top level for if `--pcr-certificate` is specified instead of `--pcr-public-key`

Reply:
Yeah I know, but I think we can consider this pcrsig stuff pretty much experimental still, as I doubt anybody but us is using it, so we I think we can still fix the public interfaces

Reply:
especially in combination with engine/provider, as that's where the break is, I would be really really surprised if anybody else was using it

Reply:
From my perspective I'm ok with this, but I'd like others to chime in here as well

Reply:
Actually, I'm not sure I'm OK with it, it will break mkosi, we depend on this behavior already.

Reply:
> Actually, I'm not sure I'm OK with it, it will break mkosi, we depend on this behavior already.

That's payback for all the times mkosi broke stuff!